Pokedex/Detail/DetailViewModel.swift

```diff
@@ -0,0 +1,44 @@
+//
+//  DetailViewModel.swift
+//  Pokedex
+//
+//  Created by krikaz on 6/25/20.
+//  Copyright © 2020 thewire. All rights reserved.
+//
+
+import UIKit
+
+final class DetailViewModel {
+    let client = PokemonClient()
+    
+    var name: String?
+    var types: String?
+    var attacks: String?
+    var pokemonSprite: ((UIImage) -> Void)?
+    var spriteURL: String? {
+        didSet {
+            DispatchQueue.main.async {
+                guard let spriteURL = self.spriteURL else { return }
+                self.client.fetchPokemonSprite(with: spriteURL) { (result) in
+                    if let data = try? result.get() {
+                        self.pokemonSprite?(UIImage(data: data)!)
+                    }
+                }
+            }
+        }
+    }
+    
+    
+    func viewDidLoad(pokemon: String?) {
```

Comment:
By moving pokemon as a property of your VM, you don't need to pass an argument to your viewDidLoad method but use the instance property.